src/codewhisperer/service/keyStrokeHandler.ts

```diff
@@ -42,6 +42,27 @@ export class KeyStrokeHandler {
         return (this.#instance ??= new this())
     }
 
+    private startIdleTimeTriggerTimer(
+        event: vscode.TextDocumentChangeEvent,
+        editor: vscode.TextEditor,
+        client: DefaultCodeWhispererClient,
+        config: ConfigurationEntry
+    ) {
+        if (this.idleTriggerTimer) {
+            clearInterval(this.idleTriggerTimer)
+            this.idleTriggerTimer = undefined
+        }
+        this.idleTriggerTimer = setInterval(() => {
+            const duration = Math.floor((performance.now() - RecommendationHandler.instance.lastInvocationTime) / 1000)
```

Comment:
is `RecommendationHandler.instance.lastInvocationTime ` accurate to describe the last user input time?

Reply:
this is not for input time, this is to ensure 2s since the lastInvocation, user input is handed by key stroke, so if there is more key stoke we cancel previous interval.